common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationFragment.java

```diff
@@ -155,17 +157,19 @@ user commit() rather than commitNow() that the fragment manager that we were rem
      * @param state a bundle containing data provided when the activity was created
      */
     void extractState(@NonNull final Bundle state) {
-        setDiagnosticContextForNewThread(state.getString(DiagnosticContext.CORRELATION_ID));
+        setDiagnosticContextForNewThread(state.getString(DiagnosticContext.CORRELATION_ID), state.getString(PRODUCT), state.getString(VERSION));
     }
 
     /**
      * When authorization fragment is launched.  It will be launched on a new thread. (TODO: verify this)
      * Initialize based on value provided in intent.
      */
-    private static void setDiagnosticContextForNewThread(final String correlationId) {
+    private static void setDiagnosticContextForNewThread(final String correlationId, final String product, final String productVersion) {
         final String methodTag = TAG + ":setDiagnosticContextForAuthorizationActivity";
         final RequestContext rc = new RequestContext();
         rc.put(DiagnosticContext.CORRELATION_ID, correlationId);
+        rc.put(PRODUCT, product);
```

Comment:
(imo, this should be part of the request object. 

DiagnosticContext is a global(static) object **_which could be modified by concurrent threads_**, and we should avoid using it as much as possible.

IIRC, today we use it for thread id and correlation id to avoid passing those value as a parameter every time we invoke a log function)

Reply:
This isn't really needed IIRC. I made this change to bring it to parity with the rest of our code base. Basically, in the thread used for authorization request, the product (library name) and version were not set in Diagnostic context, despite being set in the thread that launched the acquire token request.

Yes the original use case was for correlation id and thread id, but the product name and version were already added there for other purposes (I think Linux related primarily)

Reply:
I'll revert this addition since i'm not using it in my implementation